File: clojure/test/project_euler/sol/p0047_test.clj
(ns project-euler.sol.p0047-test
  (:require
   [clojure.test :refer [deftest testing is]]
   [project-euler.sol.p0047 :refer [solve]]))

(deftest test-p0047
  (testing "Two consecutive numbers."
    (is (= 14 (solve 2))))
  (testing "Three consecutive numbers."
    (is (= 644 (solve 3))))
  (testing "Four consecutive numbers."
    (is (= 134043 (solve 4)))))

File: go/internal/sol/p0047/p0047_test.go
package p0047

import "testing"

func Test_compute(t *testing.T) {
	type args struct {
		length int
	}
	tests := []struct {
		name string
		args args
		want string
	}{
		{name: "two consecutive numbers", args: args{2}, want: "14"},
		{name: "three consecutive numbersr", args: args{3}, want: "644"},
		{name: "four consecutive numbers", args: args{4}, want: "134043"},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := compute(tt.args.length); got != tt.want {
				t.Errorf("compute() = %v, want %v", got, tt.want)
			}
		})
	}
}
